source/fleet_provisioning.c

```diff
@@ -295,15 +295,14 @@ static FleetProvisioningStatus_t GetRegisterThingTopicCheckParams( const char *
     {
         ret = FleetProvisioningBadParameter;
 
-        LogError( ( "Invalid input parameter. pTopicBuffer: %p, bufferLength: %u, format: %d, "
-                    "topic: %d, pTemplateName: %p, templateNameLength: %u, pOutLength: %p.",
-                    ( void * ) pTopicBuffer,
-                    ( unsigned int ) bufferLength,
+        LogError( ( "Invalid input parameter. pTopicBuffer: %p, format: %d, topic: %d,"
+                    " pTemplateName: %p, templateNameLength: %u, pOutLength: %p.",
+                    ( const void * ) pTopicBuffer,
                     ( int ) format,
                     ( int ) topic,
-                    ( void * ) pTemplateName,
+                    ( const void * ) pTemplateName,
                     ( unsigned int ) templateNameLength,
-                    ( void * ) pOutLength ) );
+                    ( const void * ) pOutLength ) );
```

Comment:
Why is `pOutLength` a pointer pointing to `const` memory? Should `pOutLength` be passed as a parameter instead instead of a pointer?

Reply:
The function checks if the parameters of another function. It needs to be a pointer as we are checking if it is void. We do not modify the contents in the checking function though, so it is const.

Reply:
The semantics of this parameter are confusing, but it's okay as it is only an internal function